csi-controller SIGSEGV

<!-- This form is for bug reports and feature requests! -->

/kind bug

**What happened**:
We're trying to setup the VSphere CPI/CSI following this [doc](https://cloud-provider-vsphere.sigs.k8s.io/tutorials/kubernetes-on-vsphere-with-kubeadm.html)

The csi-controller daemonset is in a CrashLoopBackOff state. The liveness check fails, because the csi-controller container cannot start due to a SIGSEGV.

```
kubectl logs -n kube-system vsphere-csi-controller-0 vsphere-csi-controller 
I1206 07:39:54.069404       1 config.go:261] GetCnsconfig called with cfgPath: /etc/cloud/csi-vsphere.conf
I1206 07:39:54.069647       1 config.go:206] Initializing vc server 160.98.220.50
I1206 07:39:54.069698       1 controller.go:67] Initializing CNS controller
I1206 07:39:54.069734       1 virtualcentermanager.go:63] Initializing defaultVirtualCenterManager...
I1206 07:39:54.069758       1 virtualcentermanager.go:65] Successfully initialized defaultVirtualCenterManager
I1206 07:39:54.069791       1 virtualcentermanager.go:107] Successfully registered VC "160.98.220.50"
I1206 07:39:54.069819       1 manager.go:60] Initializing volume.volumeManager...
I1206 07:39:54.069842       1 manager.go:64] volume.volumeManager initialized
time="2019-12-06T07:40:15Z" level=info msg="received signal; shutting down" signal=terminated
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x128 pc=0x867dc7]

goroutine 24 [running]:
google.golang.org/grpc.(*Server).GracefulStop(0x0)
        /go/pkg/mod/google.golang.org/grpc@v1.23.0/server.go:1393 +0x37
github.com/rexray/gocsi.(*StoragePlugin).GracefulStop.func1()
        /go/pkg/mod/github.com/rexray/gocsi@v1.0.0/gocsi.go:333 +0x35
sync.(*Once).Do(0xc0002e080c, 0xc0003eeef8)
        /usr/local/go/src/sync/once.go:44 +0xb3
github.com/rexray/gocsi.(*StoragePlugin).GracefulStop(0xc0002e0780, 0x21183a0, 0xc0000ae010)
        /go/pkg/mod/github.com/rexray/gocsi@v1.0.0/gocsi.go:332 +0x56
github.com/rexray/gocsi.Run.func3()
        /go/pkg/mod/github.com/rexray/gocsi@v1.0.0/gocsi.go:121 +0x4e
github.com/rexray/gocsi.trapSignals.func1(0xc000437380, 0xc0004737a0, 0xc000473770)
        /go/pkg/mod/github.com/rexray/gocsi@v1.0.0/gocsi.go:502 +0x143
created by github.com/rexray/gocsi.trapSignals
        /go/pkg/mod/github.com/rexray/gocsi@v1.0.0/gocsi.go:487 +0x107
```
**vsphere-csi-controller-0**
```
kubectl describe pod -n kube-system vsphere-csi-controller-0 
Name:         vsphere-csi-controller-0
Namespace:    kube-system
Priority:     0
Node:         node1/160.98.236.80
Start Time:   Fri, 06 Dec 2019 07:31:50 +0000
Labels:       app=vsphere-csi-controller
              controller-revision-hash=vsphere-csi-controller-78bb4df5f7
              role=vsphere-csi
              statefulset.kubernetes.io/pod-name=vsphere-csi-controller-0
Annotations:  <none>
Status:       Running
IP:           10.245.0.5
IPs:
  IP:           10.245.0.5
Controlled By:  StatefulSet/vsphere-csi-controller
Containers:
  csi-attacher:
    Container ID:  docker://286d6b04f3cf300256681855bcd1f98903cb01d8c2da627b5952f1c642c34dae
    Image:         quay.io/k8scsi/csi-attacher:v1.1.1
    Image ID:      docker-pullable://quay.io/k8scsi/csi-attacher@sha256:e4db94969e1d463807162a1115192ed70d632a61fbeb3bdc97b40fe9ce78c831
    Port:          <none>
    Host Port:     <none>
    Args:
      --v=4
      --timeout=300s
      --csi-address=$(ADDRESS)
    State:          Running
      Started:      Fri, 06 Dec 2019 07:31:51 +0000
    Ready:          True
    Restart Count:  0
    Environment:
      ADDRESS:  /csi/csi.sock
    Mounts:
      /csi from socket-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from vsphere-csi-controller-token-h9bqc (ro)
  vsphere-csi-controller:
    Container ID:  docker://2b79e2f2a63ede8245da723f90c4f9b6e4cabdcb204d3843be01c1c3f1ec8bbf
    Image:         gcr.io/cloud-provider-vsphere/csi/release/driver:v1.0.1
    Image ID:      docker-pullable://gcr.io/cloud-provider-vsphere/csi/release/driver@sha256:fae6806f5423a0099cdf60cf53cff474b228ee4846a242d025e4833a66f91b3f
    Port:          9808/TCP
    Host Port:     0/TCP
    Args:
      --v=4
    State:          Waiting
      Reason:       CrashLoopBackOff
    Last State:     Terminated
      Reason:       Error
      Exit Code:    2
      Started:      Fri, 06 Dec 2019 07:36:47 +0000
      Finished:     Fri, 06 Dec 2019 07:37:10 +0000
    Ready:          False
    Restart Count:  7
    Liveness:       http-get http://:healthz/healthz delay=10s timeout=3s period=5s #success=1 #failure=3
    Environment:
      CSI_ENDPOINT:        unix:///var/lib/csi/sockets/pluginproxy/csi.sock
      X_CSI_MODE:          controller
      VSPHERE_CSI_CONFIG:  /etc/cloud/csi-vsphere.conf
    Mounts:
      /etc/cloud from vsphere-config-volume (ro)
      /var/lib/csi/sockets/pluginproxy/ from socket-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from vsphere-csi-controller-token-h9bqc (ro)
  liveness-probe:
    Container ID:  docker://dd27a57a7e3d3d9350685c6505498237dc6fc101e8a7dcf6af35b0cd99ce7d92
    Image:         quay.io/k8scsi/livenessprobe:v1.1.0
    Image ID:      docker-pullable://quay.io/k8scsi/livenessprobe@sha256:dde617756e0f602adc566ab71fd885f1dad451ad3fb063ac991c95a2ff47aea5
    Port:          <none>
    Host Port:     <none>
    Args:
      --csi-address=$(ADDRESS)
    State:          Running
      Started:      Fri, 06 Dec 2019 07:31:53 +0000
    Ready:          True
    Restart Count:  0
    Environment:
      ADDRESS:  /var/lib/csi/sockets/pluginproxy/csi.sock
    Mounts:
      /var/lib/csi/sockets/pluginproxy/ from socket-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from vsphere-csi-controller-token-h9bqc (ro)
  vsphere-syncer:
    Container ID:  docker://cea6e98a429f7deb145ef885ddf3238a23d6eeb595e164107c3ddf75f3b9341a
    Image:         gcr.io/cloud-provider-vsphere/csi/release/syncer:v1.0.1
    Image ID:      docker-pullable://gcr.io/cloud-provider-vsphere/csi/release/syncer@sha256:fc80ec77a2ab4b58ddfa259a938f6d741933566011d56e5ffcc8680cc83538fe
    Port:          <none>
    Host Port:     <none>
    Args:
      --v=2
    State:          Waiting
      Reason:       CrashLoopBackOff
    Last State:     Terminated
      Reason:       Error
      Exit Code:    1
      Started:      Fri, 06 Dec 2019 07:37:12 +0000
      Finished:     Fri, 06 Dec 2019 07:37:42 +0000
    Ready:          False
    Restart Count:  5
    Environment:
      FULL_SYNC_INTERVAL_MINUTES:  30
      VSPHERE_CSI_CONFIG:          /etc/cloud/csi-vsphere.conf
    Mounts:
      /etc/cloud from vsphere-config-volume (ro)
      /var/run/secrets/kubernetes.io/serviceaccount from vsphere-csi-controller-token-h9bqc (ro)
  csi-provisioner:
    Container ID:  docker://c3f788e2030f386d71252b2c63b50e1992f46eef2a4e0675cf856997d12dde2e
    Image:         quay.io/k8scsi/csi-provisioner:v1.2.2
    Image ID:      docker-pullable://quay.io/k8scsi/csi-provisioner@sha256:e3239de37c06d2bcd0e9e9648fe9a8b418d5caf9e89f243c649ff2394d3cbfef
    Port:          <none>
    Host Port:     <none>
    Args:
      --v=4
      --timeout=300s
      --csi-address=$(ADDRESS)
      --feature-gates=Topology=true
      --strict-topology
    State:          Waiting
      Reason:       CrashLoopBackOff
    Last State:     Terminated
      Reason:       Error
      Message:      Lost connection to CSI driver, exiting
      Exit Code:    255
      Started:      Fri, 06 Dec 2019 07:36:47 +0000
      Finished:     Fri, 06 Dec 2019 07:37:08 +0000
    Ready:          False
    Restart Count:  5
    Environment:
      ADDRESS:  /csi/csi.sock
    Mounts:
      /csi from socket-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from vsphere-csi-controller-token-h9bqc (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             False 
  ContainersReady   False 
  PodScheduled      True 
Volumes:
  vsphere-config-volume:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  vsphere-config-secret
    Optional:    false
  socket-dir:
    Type:          HostPath (bare host directory volume)
    Path:          /var/lib/csi/sockets/pluginproxy/csi.vsphere.vmware.com
    HostPathType:  DirectoryOrCreate
  vsphere-csi-controller-token-h9bqc:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  vsphere-csi-controller-token-h9bqc
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  node-role.kubernetes.io/master=
Tolerations:     node-role.kubernetes.io/master:NoSchedule
                 node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type     Reason     Age                    From               Message
  ----     ------     ----                   ----               -------
  Normal   Scheduled  7m17s                  default-scheduler  Successfully assigned kube-system/vsphere-csi-controller-0 to node1
  Normal   Pulled     7m15s                  kubelet, node1     Container image "quay.io/k8scsi/csi-attacher:v1.1.1" already present on machine
  Normal   Created    7m15s                  kubelet, node1     Created container csi-attacher
  Normal   Started    7m15s                  kubelet, node1     Started container csi-attacher
  Normal   Pulled     7m13s                  kubelet, node1     Container image "quay.io/k8scsi/livenessprobe:v1.1.0" already present on machine
  Normal   Pulling    7m13s                  kubelet, node1     Pulling image "gcr.io/cloud-provider-vsphere/csi/release/syncer:v1.0.1"
  Normal   Created    7m13s                  kubelet, node1     Created container liveness-probe
  Normal   Started    7m13s                  kubelet, node1     Started container liveness-probe
  Normal   Started    7m12s                  kubelet, node1     Started container vsphere-syncer
  Normal   Pulled     7m12s                  kubelet, node1     Container image "quay.io/k8scsi/csi-provisioner:v1.2.2" already present on machine
  Normal   Pulled     7m12s                  kubelet, node1     Successfully pulled image "gcr.io/cloud-provider-vsphere/csi/release/syncer:v1.0.1"
  Normal   Created    7m12s                  kubelet, node1     Created container vsphere-syncer
  Normal   Created    7m11s                  kubelet, node1     Created container csi-provisioner
  Normal   Started    7m11s                  kubelet, node1     Started container csi-provisioner
  Normal   Pulling    6m51s (x2 over 7m15s)  kubelet, node1     Pulling image "gcr.io/cloud-provider-vsphere/csi/release/driver:v1.0.1"
  Normal   Killing    6m51s                  kubelet, node1     Container vsphere-csi-controller failed liveness probe, will be restarted
  Normal   Started    6m50s (x2 over 7m13s)  kubelet, node1     Started container vsphere-csi-controller
  Normal   Created    6m50s (x2 over 7m14s)  kubelet, node1     Created container vsphere-csi-controller
  Normal   Pulled     6m50s (x2 over 7m14s)  kubelet, node1     Successfully pulled image "gcr.io/cloud-provider-vsphere/csi/release/driver:v1.0.1"
  Warning  Unhealthy  2m6s (x22 over 7m1s)   kubelet, node1     Liveness probe failed: Get http://10.245.0.5:9808/healthz: dial tcp 10.245.0.5:9808: connect: connection refused
```
**What you expected to happen**:
The daemonset should start successfully
**How to reproduce it (as minimally and precisely as possible)**:
follow the documentation in the link up-there

**Anything else we need to know?**:

**Environment**:
- vsphere-cloud-controller-manager version: 1.0.1
- OS (e.g. from /etc/os-release): Ubuntu 18.04.3 LTS, Kubernetes v1.16.3
- Kernel (e.g. `uname -a`): 4.15.0-72-generic
- Install tools: kubeadm
- Others: vCenter Appliance 6.7 Update 3 (6.7.0.40000)

Kube Version:
```
[root@testvm-master ~]# kubectl version
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.2", GitCommit:"59603c6e503c87169aea6106f57b9f242f64df89", GitTreeState:"clean", BuildDate:"2020-01-18T23:30:10Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.2", GitCommit:"59603c6e503c87169aea6106f57b9f242f64df89", GitTreeState:"clean", BuildDate:"2020-01-18T23:22:30Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"linux/amd64"}
```
Controller container versions:
```
[root@testvm-master ~]# kubectl -n kube-system describe pod/vsphere-csi-controller-0
Containers:
  csi-attacher:
    Container ID:  docker://d86c533586966d1b2b30cbcc3f2f06ae782b84d424601c3dae253afc8b24485e
    Image:         quay.io/k8scsi/csi-attacher:v1.1.1
    Image ID:      docker-pullable://quay.io/k8scsi/csi-attacher@sha256:e4db94969e1d463807162a1115192ed70d632a61fbeb3bdc97b40fe9ce78c831
...
  vsphere-csi-controller:
    Container ID:  docker://1ffe53f76d5643d8381a2fd094066964d6fca282c2477b08d9b708069b00e978
    Image:         gcr.io/cloud-provider-vsphere/csi/release/driver:v1.0.1
    Image ID:      docker-pullable://gcr.io/cloud-provider-vsphere/csi/release/driver@sha256:fae6806f5423a0099cdf60cf53cff474b228ee4846a242d025e4833a66f91b3f
...
  liveness-probe:
    Container ID:  docker://c8600f410c772f587df1af3234c0e9f6c9d14efa2154fd98385702e8eaa8b039
    Image:         quay.io/k8scsi/livenessprobe:v1.1.0
    Image ID:      docker-pullable://quay.io/k8scsi/livenessprobe@sha256:dde617756e0f602adc566ab71fd885f1dad451ad3fb063ac991c95a2ff47aea5
...
  vsphere-syncer:
    Container ID:  docker://f54be7525e42780463ec717c282c4e2071420ed27feacb8962ce0cab346c4a91
    Image:         gcr.io/cloud-provider-vsphere/csi/release/syncer:v1.0.1
    Image ID:      docker-pullable://gcr.io/cloud-provider-vsphere/csi/release/syncer@sha256:fc80ec77a2ab4b58ddfa259a938f6d741933566011d56e5ffcc8680cc83538fe
...
  csi-provisioner:
    Container ID:  docker://8cc7adb104527be5b94812fb15e7c016e214a96d5c08efe08f030d7e71861e3e
    Image:         quay.io/k8scsi/csi-provisioner:v1.2.2
    Image ID:      docker-pullable://quay.io/k8scsi/csi-provisioner@sha256:e3239de37c06d2bcd0e9e9648fe9a8b418d5caf9e89f243c649ff2394d3cbfef
...

```
Controller Log: 
```
[root@testvm-master ~]# kubectl -n kube-system logs pod/vsphere-csi-controller-0 vsphere-csi-controller
I0303 14:05:38.362860       1 config.go:261] GetCnsconfig called with cfgPath: /etc/cloud/csi-vsphere.conf
I0303 14:05:38.363067       1 config.go:206] Initializing vc server x.x.x.180
I0303 14:05:38.363084       1 controller.go:67] Initializing CNS controller
I0303 14:05:38.363100       1 virtualcentermanager.go:63] Initializing defaultVirtualCenterManager...
I0303 14:05:38.363104       1 virtualcentermanager.go:65] Successfully initialized defaultVirtualCenterManager
I0303 14:05:38.363116       1 virtualcentermanager.go:107] Successfully registered VC "x.x.x.180"
I0303 14:05:38.363123       1 manager.go:60] Initializing volume.volumeManager...
I0303 14:05:38.363127       1 manager.go:64] volume.volumeManager initialized
time="2020-03-03T14:06:01Z" level=info msg="received signal; shutting down" signal=terminated
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x128 pc=0x867dc7]

goroutine 20 [running]:
google.golang.org/grpc.(*Server).GracefulStop(0x0)
        /go/pkg/mod/google.golang.org/grpc@v1.23.0/server.go:1393 +0x37
github.com/rexray/gocsi.(*StoragePlugin).GracefulStop.func1()
        /go/pkg/mod/github.com/rexray/gocsi@v1.0.0/gocsi.go:333 +0x35
sync.(*Once).Do(0xc0002e680c, 0xc0004caef8)
        /usr/local/go/src/sync/once.go:44 +0xb3
github.com/rexray/gocsi.(*StoragePlugin).GracefulStop(0xc0002e6780, 0x21183a0, 0xc000056018)
        /go/pkg/mod/github.com/rexray/gocsi@v1.0.0/gocsi.go:332 +0x56
github.com/rexray/gocsi.Run.func3()
        /go/pkg/mod/github.com/rexray/gocsi@v1.0.0/gocsi.go:121 +0x4e
github.com/rexray/gocsi.trapSignals.func1(0xc00009c540, 0xc0000bdb90, 0xc0000bdad0)
        /go/pkg/mod/github.com/rexray/gocsi@v1.0.0/gocsi.go:502 +0x143
created by github.com/rexray/gocsi.trapSignals
        /go/pkg/mod/github.com/rexray/gocsi@v1.0.0/gocsi.go:487 +0x107
```
Syncer Log:
```
[root@testvm-master ~]# kubectl -n kube-system logs pod/vsphere-csi-controller-0 vsphere-syncer
I0303 14:02:24.295245       1 virtualcentermanager.go:63] Initializing defaultVirtualCenterManager...
I0303 14:02:24.295340       1 virtualcentermanager.go:65] Successfully initialized defaultVirtualCenterManager
I0303 14:02:24.295367       1 virtualcentermanager.go:107] Successfully registered VC "x.x.x.180"
E0303 14:02:54.296310       1 virtualcenter.go:116] Failed to create new client with err: Post https://x.x.x.180:443/sdk: dial tcp x.x.x.180:443: i/o timeout
E0303 14:02:54.296404       1 virtualcenter.go:227] Failed to create govmomi client with err: Post https://x.x.x.180:443/sdk: dial tcp x.x.x.180:443: i/o timeout
E0303 14:02:54.296426       1 virtualcenter.go:190] Cannot connect to vCenter with err: Post https://x.x.x.180:443/sdk: dial tcp x.x.x.180:443: i/o timeout
E0303 14:02:54.296443       1 metadatasyncer.go:105] Failed to connect to VirtualCenter host: "x.x.x.180". err=Post https://x.x.x.180:443/sdk: dial tcp x.x.x.180:443: i/o timeout
E0303 14:02:54.296464       1 main.go:34] Error initializing Metadata Syncer
```
Other Info:
```
[root@testvm-master ~]# kubectl get nodes
NAME                             STATUS   ROLES    AGE   VERSION
testvm-master.x.x.x.x   Ready    master   33d   v1.17.2
testvm-node0.x.x.x.x    Ready    <none>   33d   v1.17.2
testvm-node1.x.x.x.x    Ready    <none>   33d   v1.17.2
```
```
[root@testvm-master ~]# kubectl -n kube-system describe statefulset.apps/vsphere-csi-controller
Name:               vsphere-csi-controller
Namespace:          kube-system
CreationTimestamp:  Wed, 29 Jan 2020 21:14:30 +0000
Selector:           app=vsphere-csi-controller
Labels:             <none>
Annotations:        kubectl.kubernetes.io/last-applied-configuration:
                      {"apiVersion":"apps/v1","kind":"StatefulSet","metadata":{"annotations":{},"name":"vsphere-csi-controller","namespace":"kube-system"},"spec...
Replicas:           1 desired | 1 total
Update Strategy:    RollingUpdate
Pods Status:        1 Running / 0 Waiting / 0 Succeeded / 0 Failed
Pod Template:
  Labels:           app=vsphere-csi-controller
                    role=vsphere-csi
  Service Account:  vsphere-csi-controller
  Containers:
   csi-attacher:
    Image:      quay.io/k8scsi/csi-attacher:v1.1.1
    Port:       <none>
    Host Port:  <none>
    Args:
      --v=4
      --timeout=300s
      --csi-address=$(ADDRESS)
    Environment:
      ADDRESS:  /csi/csi.sock
    Mounts:
      /csi from socket-dir (rw)
   vsphere-csi-controller:
    Image:      gcr.io/cloud-provider-vsphere/csi/release/driver:v1.0.1
    Port:       9808/TCP
    Host Port:  0/TCP
    Args:
      --v=4
    Liveness:  http-get http://:healthz/healthz delay=10s timeout=3s period=5s #success=1 #failure=3
    Environment:
      CSI_ENDPOINT:        unix:///var/lib/csi/sockets/pluginproxy/csi.sock
      X_CSI_MODE:          controller
      VSPHERE_CSI_CONFIG:  /etc/cloud/csi-vsphere.conf
    Mounts:
      /etc/cloud from vsphere-config-volume (ro)
      /var/lib/csi/sockets/pluginproxy/ from socket-dir (rw)
   liveness-probe:
    Image:      quay.io/k8scsi/livenessprobe:v1.1.0
    Port:       <none>
    Host Port:  <none>
    Args:
      --csi-address=$(ADDRESS)
    Environment:
      ADDRESS:  /var/lib/csi/sockets/pluginproxy/csi.sock
    Mounts:
      /var/lib/csi/sockets/pluginproxy/ from socket-dir (rw)
   vsphere-syncer:
    Image:      gcr.io/cloud-provider-vsphere/csi/release/syncer:v1.0.1
    Port:       <none>
    Host Port:  <none>
    Args:
      --v=2
    Environment:
      FULL_SYNC_INTERVAL_MINUTES:  30
      VSPHERE_CSI_CONFIG:          /etc/cloud/csi-vsphere.conf
    Mounts:
      /etc/cloud from vsphere-config-volume (ro)
   csi-provisioner:
    Image:      quay.io/k8scsi/csi-provisioner:v1.2.2
    Port:       <none>
    Host Port:  <none>
    Args:
      --v=4
      --timeout=300s
      --csi-address=$(ADDRESS)
      --feature-gates=Topology=true
      --strict-topology
    Environment:
      ADDRESS:  /csi/csi.sock
    Mounts:
      /csi from socket-dir (rw)
  Volumes:
   vsphere-config-volume:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  vsphere-config-secret
    Optional:    false
   socket-dir:
    Type:          HostPath (bare host directory volume)
    Path:          /var/lib/csi/sockets/pluginproxy/csi.vsphere.vmware.com
    HostPathType:  DirectoryOrCreate
Volume Claims:     <none>
Events:            <none>
```
Also I've basically left it running since I posted my issue :)
```
[root@testvm-master ~]# kubectl -n kube-system get all
NAME                                                         READY   STATUS             RESTARTS   AGE
pod/coredns-6955765f44-4nmxj                                 1/1     Running            1          33d
pod/coredns-6955765f44-xbxvp                                 1/1     Running            1          33d
pod/etcd-testvm-master.x.x.x.x                      1/1     Running            1          33d
pod/kube-apiserver-testvm-master.x.x.x.x            1/1     Running            1          33d
pod/kube-controller-manager-testvm-master.x.x.x.x   1/1     Running            1          33d
pod/kube-flannel-ds-amd64-l7qlh                              1/1     Running            2          33d
pod/kube-flannel-ds-amd64-nqdj9                              1/1     Running            1          33d
pod/kube-flannel-ds-amd64-xlkjs                              1/1     Running            3          33d
pod/kube-proxy-bnt9b                                         1/1     Running            3          33d
pod/kube-proxy-g8rxv                                         1/1     Running            2          33d
pod/kube-proxy-nqm5p                                         1/1     Running            1          33d
pod/kube-scheduler-testvm-master.x.x.x.x            1/1     Running            1          33d
pod/vsphere-cloud-controller-manager-pplht                   1/1     Running            1          33d
pod/vsphere-csi-controller-0                                 3/5     CrashLoopBackOff   25514      33d
pod/vsphere-csi-node-q2dnc                                   3/3     Running            9          33d
pod/vsphere-csi-node-qs457                                   3/3     Running            6          33d

NAME                                       TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)                  AGE
service/kube-dns                           ClusterIP   10.96.0.10       <none>        53/UDP,53/TCP,9153/TCP   33d
service/vsphere-cloud-controller-manager   NodePort    10.105.209.218   <none>        43001:30133/TCP          33d

NAME                                              DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR                     AGE
daemonset.apps/kube-flannel-ds-amd64              3         3         3       3            3           <none>                            33d
daemonset.apps/kube-flannel-ds-arm                0         0         0       0            0           <none>                            33d
daemonset.apps/kube-flannel-ds-arm64              0         0         0       0            0           <none>                            33d
daemonset.apps/kube-flannel-ds-ppc64le            0         0         0       0            0           <none>                            33d
daemonset.apps/kube-flannel-ds-s390x              0         0         0       0            0           <none>                            33d
daemonset.apps/kube-proxy                         3         3         3       3            3           beta.kubernetes.io/os=linux       33d
daemonset.apps/vsphere-cloud-controller-manager   1         1         1       1            1           node-role.kubernetes.io/master=   33d
daemonset.apps/vsphere-csi-node                   2         2         2       2            2           <none>                            33d

NAME                      READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/coredns   2/2     2            2           33d

NAME                                 DESIRED   CURRENT   READY   AGE
replicaset.apps/coredns-6955765f44   2         2         2       33d

NAME                                      READY   AGE
statefulset.apps/vsphere-csi-controller   0/1     33d
```

Kube Version:
```
[root@testvm-master ~]# kubectl version
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.2", GitCommit:"59603c6e503c87169aea6106f57b9f242f64df89", GitTreeState:"clean", BuildDate:"2020-01-18T23:30:10Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.2", GitCommit:"59603c6e503c87169aea6106f57b9f242f64df89", GitTreeState:"clean", BuildDate:"2020-01-18T23:22:30Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"linux/amd64"}
```
Controller container versions:
```
[root@testvm-master ~]# kubectl -n kube-system describe pod/vsphere-csi-controller-0
Containers:
  csi-attacher:
    Container ID:  docker://d86c533586966d1b2b30cbcc3f2f06ae782b84d424601c3dae253afc8b24485e
    Image:         quay.io/k8scsi/csi-attacher:v1.1.1
    Image ID:      docker-pullable://quay.io/k8scsi/csi-attacher@sha256:e4db94969e1d463807162a1115192ed70d632a61fbeb3bdc97b40fe9ce78c831
...
  vsphere-csi-controller:
    Container ID:  docker://1ffe53f76d5643d8381a2fd094066964d6fca282c2477b08d9b708069b00e978
    Image:         gcr.io/cloud-provider-vsphere/csi/release/driver:v1.0.1
    Image ID:      docker-pullable://gcr.io/cloud-provider-vsphere/csi/release/driver@sha256:fae6806f5423a0099cdf60cf53cff474b228ee4846a242d025e4833a66f91b3f
...
  liveness-probe:
    Container ID:  docker://c8600f410c772f587df1af3234c0e9f6c9d14efa2154fd98385702e8eaa8b039
    Image:         quay.io/k8scsi/livenessprobe:v1.1.0
    Image ID:      docker-pullable://quay.io/k8scsi/livenessprobe@sha256:dde617756e0f602adc566ab71fd885f1dad451ad3fb063ac991c95a2ff47aea5
...
  vsphere-syncer:
    Container ID:  docker://f54be7525e42780463ec717c282c4e2071420ed27feacb8962ce0cab346c4a91
    Image:         gcr.io/cloud-provider-vsphere/csi/release/syncer:v1.0.1
    Image ID:      docker-pullable://gcr.io/cloud-provider-vsphere/csi/release/syncer@sha256:fc80ec77a2ab4b58ddfa259a938f6d741933566011d56e5ffcc8680cc83538fe
...
  csi-provisioner:
    Container ID:  docker://8cc7adb104527be5b94812fb15e7c016e214a96d5c08efe08f030d7e71861e3e
    Image:         quay.io/k8scsi/csi-provisioner:v1.2.2
    Image ID:      docker-pullable://quay.io/k8scsi/csi-provisioner@sha256:e3239de37c06d2bcd0e9e9648fe9a8b418d5caf9e89f243c649ff2394d3cbfef
...

```
Controller Log: 
```
[root@testvm-master ~]# kubectl -n kube-system logs pod/vsphere-csi-controller-0 vsphere-csi-controller
I0303 14:05:38.362860       1 config.go:261] GetCnsconfig called with cfgPath: /etc/cloud/csi-vsphere.conf
I0303 14:05:38.363067       1 config.go:206] Initializing vc server x.x.x.180
I0303 14:05:38.363084       1 controller.go:67] Initializing CNS controller
I0303 14:05:38.363100       1 virtualcentermanager.go:63] Initializing defaultVirtualCenterManager...
I0303 14:05:38.363104       1 virtualcentermanager.go:65] Successfully initialized defaultVirtualCenterManager
I0303 14:05:38.363116       1 virtualcentermanager.go:107] Successfully registered VC "x.x.x.180"
I0303 14:05:38.363123       1 manager.go:60] Initializing volume.volumeManager...
I0303 14:05:38.363127       1 manager.go:64] volume.volumeManager initialized
time="2020-03-03T14:06:01Z" level=info msg="received signal; shutting down" signal=terminated
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x128 pc=0x867dc7]

goroutine 20 [running]:
google.golang.org/grpc.(*Server).GracefulStop(0x0)
        /go/pkg/mod/google.golang.org/grpc@v1.23.0/server.go:1393 +0x37
github.com/rexray/gocsi.(*StoragePlugin).GracefulStop.func1()
        /go/pkg/mod/github.com/rexray/gocsi@v1.0.0/gocsi.go:333 +0x35
sync.(*Once).Do(0xc0002e680c, 0xc0004caef8)
        /usr/local/go/src/sync/once.go:44 +0xb3
github.com/rexray/gocsi.(*StoragePlugin).GracefulStop(0xc0002e6780, 0x21183a0, 0xc000056018)
        /go/pkg/mod/github.com/rexray/gocsi@v1.0.0/gocsi.go:332 +0x56
github.com/rexray/gocsi.Run.func3()
        /go/pkg/mod/github.com/rexray/gocsi@v1.0.0/gocsi.go:121 +0x4e
github.com/rexray/gocsi.trapSignals.func1(0xc00009c540, 0xc0000bdb90, 0xc0000bdad0)
        /go/pkg/mod/github.com/rexray/gocsi@v1.0.0/gocsi.go:502 +0x143
created by github.com/rexray/gocsi.trapSignals
        /go/pkg/mod/github.com/rexray/gocsi@v1.0.0/gocsi.go:487 +0x107
```
Syncer Log:
```
[root@testvm-master ~]# kubectl -n kube-system logs pod/vsphere-csi-controller-0 vsphere-syncer
I0303 14:02:24.295245       1 virtualcentermanager.go:63] Initializing defaultVirtualCenterManager...
I0303 14:02:24.295340       1 virtualcentermanager.go:65] Successfully initialized defaultVirtualCenterManager
I0303 14:02:24.295367       1 virtualcentermanager.go:107] Successfully registered VC "x.x.x.180"
E0303 14:02:54.296310       1 virtualcenter.go:116] Failed to create new client with err: Post https://x.x.x.180:443/sdk: dial tcp x.x.x.180:443: i/o timeout
E0303 14:02:54.296404       1 virtualcenter.go:227] Failed to create govmomi client with err: Post https://x.x.x.180:443/sdk: dial tcp x.x.x.180:443: i/o timeout
E0303 14:02:54.296426       1 virtualcenter.go:190] Cannot connect to vCenter with err: Post https://x.x.x.180:443/sdk: dial tcp x.x.x.180:443: i/o timeout
E0303 14:02:54.296443       1 metadatasyncer.go:105] Failed to connect to VirtualCenter host: "x.x.x.180". err=Post https://x.x.x.180:443/sdk: dial tcp x.x.x.180:443: i/o timeout
E0303 14:02:54.296464       1 main.go:34] Error initializing Metadata Syncer
```
Other Info:
```
[root@testvm-master ~]# kubectl get nodes
NAME                             STATUS   ROLES    AGE   VERSION
testvm-master.x.x.x.x   Ready    master   33d   v1.17.2
testvm-node0.x.x.x.x    Ready    <none>   33d   v1.17.2
testvm-node1.x.x.x.x    Ready    <none>   33d   v1.17.2
```
```
[root@testvm-master ~]# kubectl -n kube-system describe statefulset.apps/vsphere-csi-controller
Name:               vsphere-csi-controller
Namespace:          kube-system
CreationTimestamp:  Wed, 29 Jan 2020 21:14:30 +0000
Selector:           app=vsphere-csi-controller
Labels:             <none>
Annotations:        kubectl.kubernetes.io/last-applied-configuration:
                      {"apiVersion":"apps/v1","kind":"StatefulSet","metadata":{"annotations":{},"name":"vsphere-csi-controller","namespace":"kube-system"},"spec...
Replicas:           1 desired | 1 total
Update Strategy:    RollingUpdate
Pods Status:        1 Running / 0 Waiting / 0 Succeeded / 0 Failed
Pod Template:
  Labels:           app=vsphere-csi-controller
                    role=vsphere-csi
  Service Account:  vsphere-csi-controller
  Containers:
   csi-attacher:
    Image:      quay.io/k8scsi/csi-attacher:v1.1.1
    Port:       <none>
    Host Port:  <none>
    Args:
      --v=4
      --timeout=300s
      --csi-address=$(ADDRESS)
    Environment:
      ADDRESS:  /csi/csi.sock
    Mounts:
      /csi from socket-dir (rw)
   vsphere-csi-controller:
    Image:      gcr.io/cloud-provider-vsphere/csi/release/driver:v1.0.1
    Port:       9808/TCP
    Host Port:  0/TCP
    Args:
      --v=4
    Liveness:  http-get http://:healthz/healthz delay=10s timeout=3s period=5s #success=1 #failure=3
    Environment:
      CSI_ENDPOINT:        unix:///var/lib/csi/sockets/pluginproxy/csi.sock
      X_CSI_MODE:          controller
      VSPHERE_CSI_CONFIG:  /etc/cloud/csi-vsphere.conf
    Mounts:
      /etc/cloud from vsphere-config-volume (ro)
      /var/lib/csi/sockets/pluginproxy/ from socket-dir (rw)
   liveness-probe:
    Image:      quay.io/k8scsi/livenessprobe:v1.1.0
    Port:       <none>
    Host Port:  <none>
    Args:
      --csi-address=$(ADDRESS)
    Environment:
      ADDRESS:  /var/lib/csi/sockets/pluginproxy/csi.sock
    Mounts:
      /var/lib/csi/sockets/pluginproxy/ from socket-dir (rw)
   vsphere-syncer:
    Image:      gcr.io/cloud-provider-vsphere/csi/release/syncer:v1.0.1
    Port:       <none>
    Host Port:  <none>
    Args:
      --v=2
    Environment:
      FULL_SYNC_INTERVAL_MINUTES:  30
      VSPHERE_CSI_CONFIG:          /etc/cloud/csi-vsphere.conf
    Mounts:
      /etc/cloud from vsphere-config-volume (ro)
   csi-provisioner:
    Image:      quay.io/k8scsi/csi-provisioner:v1.2.2
    Port:       <none>
    Host Port:  <none>
    Args:
      --v=4
      --timeout=300s
      --csi-address=$(ADDRESS)
      --feature-gates=Topology=true
      --strict-topology
    Environment:
      ADDRESS:  /csi/csi.sock
    Mounts:
      /csi from socket-dir (rw)
  Volumes:
   vsphere-config-volume:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  vsphere-config-secret
    Optional:    false
   socket-dir:
    Type:          HostPath (bare host directory volume)
    Path:          /var/lib/csi/sockets/pluginproxy/csi.vsphere.vmware.com
    HostPathType:  DirectoryOrCreate
Volume Claims:     <none>
Events:            <none>
```
Also I've basically left it running since I posted my issue :)
```
[root@testvm-master ~]# kubectl -n kube-system get all
NAME                                                         READY   STATUS             RESTARTS   AGE
pod/coredns-6955765f44-4nmxj                                 1/1     Running            1          33d
pod/coredns-6955765f44-xbxvp                                 1/1     Running            1          33d
pod/etcd-testvm-master.x.x.x.x                      1/1     Running            1          33d
pod/kube-apiserver-testvm-master.x.x.x.x            1/1     Running            1          33d
pod/kube-controller-manager-testvm-master.x.x.x.x   1/1     Running            1          33d
pod/kube-flannel-ds-amd64-l7qlh                              1/1     Running            2          33d
pod/kube-flannel-ds-amd64-nqdj9                              1/1     Running            1          33d
pod/kube-flannel-ds-amd64-xlkjs                              1/1     Running            3          33d
pod/kube-proxy-bnt9b                                         1/1     Running            3          33d
pod/kube-proxy-g8rxv                                         1/1     Running            2          33d
pod/kube-proxy-nqm5p                                         1/1     Running            1          33d
pod/kube-scheduler-testvm-master.x.x.x.x            1/1     Running            1          33d
pod/vsphere-cloud-controller-manager-pplht                   1/1     Running            1          33d
pod/vsphere-csi-controller-0                                 3/5     CrashLoopBackOff   25514      33d
pod/vsphere-csi-node-q2dnc                                   3/3     Running            9          33d
pod/vsphere-csi-node-qs457                                   3/3     Running            6          33d

NAME                                       TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)                  AGE
service/kube-dns                           ClusterIP   10.96.0.10       <none>        53/UDP,53/TCP,9153/TCP   33d
service/vsphere-cloud-controller-manager   NodePort    10.105.209.218   <none>        43001:30133/TCP          33d

NAME                                              DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR                     AGE
daemonset.apps/kube-flannel-ds-amd64              3         3         3       3            3           <none>                            33d
daemonset.apps/kube-flannel-ds-arm                0         0         0       0            0           <none>                            33d
daemonset.apps/kube-flannel-ds-arm64              0         0         0       0            0           <none>                            33d
daemonset.apps/kube-flannel-ds-ppc64le            0         0         0       0            0           <none>                            33d
daemonset.apps/kube-flannel-ds-s390x              0         0         0       0            0           <none>                            33d
daemonset.apps/kube-proxy                         3         3         3       3            3           beta.kubernetes.io/os=linux       33d
daemonset.apps/vsphere-cloud-controller-manager   1         1         1       1            1           node-role.kubernetes.io/master=   33d
daemonset.apps/vsphere-csi-node                   2         2         2       2            2           <none>                            33d

NAME                      READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/coredns   2/2     2            2           33d

NAME                                 DESIRED   CURRENT   READY   AGE
replicaset.apps/coredns-6955765f44   2         2         2       33d

NAME                                      READY   AGE
statefulset.apps/vsphere-csi-controller   0/1     33d
```

I can pretty reliably reproduce this. I'm using clusters provisioned on VSphere 6.7U3 with Rancher 2.3.5, using the instructions [on the official docs](https://cloud-provider-vsphere.sigs.k8s.io/tutorials/kubernetes-on-vsphere-with-kubeadm.html) and a couple required modifications I found searching around.

I've reproduced it with both Kubernetes 1.16 and 1.17, but for this I'll use 1.17:
```
$ kubectl version --kubeconfig csi-test-kube.yaml
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.1", GitCommit:"d224476cd0730baca2b6e357d144171ed74192d6", GitTreeState:"clean", BuildDate:"2020-01-14T21:04:32Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.2", GitCommit:"59603c6e503c87169aea6106f57b9f242f64df89", GitTreeState:"clean", BuildDate:"2020-01-18T23:22:30Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"linux/amd64"}
```

My Rancher-provisioned cluster required the modifications as described [on this forum post](https://forums.rancher.com/t/kubernetes-cluster-on-vsphere-with-csi-and-cpi/15648/4). This notably modifies the RKE config when initially provisioning, and these files from the official documentation:
- https://github.com/kubernetes/cloud-provider-vsphere/raw/master/manifests/controller-manager/vsphere-cloud-controller-manager-ds.yaml
- https://raw.githubusercontent.com/kubernetes-sigs/vsphere-csi-driver/master/manifests/1.14/deploy/vsphere-csi-controller-ss.yaml
- The csi-node yaml (https://raw.githubusercontent.com/kubernetes-sigs/vsphere-csi-driver/master/manifests/1.14/deploy/vsphere-csi-node-ds.yaml) also needs modifications to work with the RKE-provisioned Kubelet as described in [this ticket](https://github.com/kubernetes-sigs/vsphere-csi-driver/issues/124).

Beyond those modifications, I follow the vsphere cloud provider documentation exactly to set up the CPI/CSI.

Here's my csi-controller pod:
```
$ kubectl -n kube-system describe pod/vsphere-csi-controller-0
Name:         vsphere-csi-controller-0
Namespace:    kube-system
Priority:     0
Node:         hill-scs-master-1/10.16.224.216
Start Time:   Mon, 02 Mar 2020 17:11:22 -0800
Labels:       app=vsphere-csi-controller
              controller-revision-hash=vsphere-csi-controller-5c469fb9fb
              role=vsphere-csi
              statefulset.kubernetes.io/pod-name=vsphere-csi-controller-0
Annotations:  <none>
Status:       Running
IP:           10.42.0.4
IPs:
  IP:           10.42.0.4
Controlled By:  StatefulSet/vsphere-csi-controller
Containers:
  csi-attacher:
    Container ID:  docker://5a20de7808b320856cbefe329f94a2b2a9e9ac9558a2856864fac2d2fdebbaf9
    Image:         quay.io/k8scsi/csi-attacher:v1.1.1
    Image ID:      docker-pullable://quay.io/k8scsi/csi-attacher@sha256:e4db94969e1d463807162a1115192ed70d632a61fbeb3bdc97b40fe9ce78c831
    Port:          <none>
    Host Port:     <none>
    Args:
      --v=4
      --timeout=300s
      --csi-address=$(ADDRESS)
    State:          Running
      Started:      Mon, 02 Mar 2020 17:11:23 -0800
    Ready:          True
    Restart Count:  0
    Environment:
      ADDRESS:  /csi/csi.sock
    Mounts:
      /csi from socket-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from vsphere-csi-controller-token-5r228 (ro)
  vsphere-csi-controller:
    Container ID:  docker://eeec8fb8e4435d82758d4a9d5bbe4cc209593def2108148c2bcd40b34068e13e
    Image:         gcr.io/cloud-provider-vsphere/csi/release/driver:v1.0.2
    Image ID:      docker-pullable://gcr.io/cloud-provider-vsphere/csi/release/driver@sha256:149e87faaacda614ee95ec271b54c8bfdbd2bf5825abc12d45c654036b798229
    Port:          9808/TCP
    Host Port:     0/TCP
    Args:
      --v=5
    State:          Running
      Started:      Mon, 02 Mar 2020 17:12:11 -0800
    Last State:     Terminated
      Reason:       Error
      Exit Code:    2
      Started:      Mon, 02 Mar 2020 17:11:45 -0800
      Finished:     Mon, 02 Mar 2020 17:12:10 -0800
    Ready:          True
    Restart Count:  2
    Liveness:       http-get http://:healthz/healthz delay=10s timeout=3s period=5s #success=1 #failure=3
    Environment:
      CSI_ENDPOINT:        unix:///var/lib/csi/sockets/pluginproxy/csi.sock
      X_CSI_MODE:          controller
      VSPHERE_CSI_CONFIG:  /etc/cloud/csi-vsphere.conf
    Mounts:
      /etc/cloud from vsphere-config-volume (ro)
      /var/lib/csi/sockets/pluginproxy/ from socket-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from vsphere-csi-controller-token-5r228 (ro)
  liveness-probe:
    Container ID:  docker://96fb5033652a3d3fa9b86a7e024602c590e8696f325402f673c30f05b032b921
    Image:         quay.io/k8scsi/livenessprobe:v1.1.0
    Image ID:      docker-pullable://quay.io/k8scsi/livenessprobe@sha256:dde617756e0f602adc566ab71fd885f1dad451ad3fb063ac991c95a2ff47aea5
    Port:          <none>
    Host Port:     <none>
    Args:
      --csi-address=$(ADDRESS)
    State:          Running
      Started:      Mon, 02 Mar 2020 17:11:24 -0800
    Ready:          True
    Restart Count:  0
    Environment:
      ADDRESS:  /var/lib/csi/sockets/pluginproxy/csi.sock
    Mounts:
      /var/lib/csi/sockets/pluginproxy/ from socket-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from vsphere-csi-controller-token-5r228 (ro)
  vsphere-syncer:
    Container ID:  docker://e23026264b3983c313e42b1611fdeaa99616e3e6c51f1da5444a11382a1ef655
    Image:         gcr.io/cloud-provider-vsphere/csi/release/syncer:v1.0.2
    Image ID:      docker-pullable://gcr.io/cloud-provider-vsphere/csi/release/syncer@sha256:4b36adf433b98aa84a8f2e58db1b3bea8dfcb05ee56e65c9e9388d56c25e8647
    Port:          <none>
    Host Port:     <none>
    Args:
      --v=2
    State:          Running
      Started:      Mon, 02 Mar 2020 17:11:25 -0800
    Ready:          True
    Restart Count:  0
    Environment:
      FULL_SYNC_INTERVAL_MINUTES:  30
      VSPHERE_CSI_CONFIG:          /etc/cloud/csi-vsphere.conf
    Mounts:
      /etc/cloud from vsphere-config-volume (ro)
      /var/run/secrets/kubernetes.io/serviceaccount from vsphere-csi-controller-token-5r228 (ro)
  csi-provisioner:
    Container ID:  docker://6ec3a1843095d4e6c8976fab5045ed1aabbfbfbeee538b5cdebbd93c51839d99
    Image:         quay.io/k8scsi/csi-provisioner:v1.2.2
    Image ID:      docker-pullable://quay.io/k8scsi/csi-provisioner@sha256:e3239de37c06d2bcd0e9e9648fe9a8b418d5caf9e89f243c649ff2394d3cbfef
    Port:          <none>
    Host Port:     <none>
    Args:
      --v=4
      --timeout=300s
      --csi-address=$(ADDRESS)
      --feature-gates=Topology=true
      --strict-topology
    State:          Running
      Started:      Mon, 02 Mar 2020 17:12:28 -0800
    Last State:     Terminated
      Reason:       Error
      Message:      Lost connection to CSI driver, exiting
      Exit Code:    255
      Started:      Mon, 02 Mar 2020 17:11:47 -0800
      Finished:     Mon, 02 Mar 2020 17:12:08 -0800
    Ready:          True
    Restart Count:  2
    Environment:
      ADDRESS:  /csi/csi.sock
    Mounts:
      /csi from socket-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from vsphere-csi-controller-token-5r228 (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             True
  ContainersReady   True
  PodScheduled      True
Volumes:
  vsphere-config-volume:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  vsphere-config-secret
    Optional:    false
  socket-dir:
    Type:          HostPath (bare host directory volume)
    Path:          /var/lib/csi/sockets/pluginproxy/csi.vsphere.vmware.com
    HostPathType:  DirectoryOrCreate
  vsphere-csi-controller-token-5r228:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  vsphere-csi-controller-token-5r228
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  node-role.kubernetes.io/controlplane=true
Tolerations:     node-role.kubernetes.io/controlplane=true:NoSchedule
                 node-role.kubernetes.io/etcd=true:NoExecute
                 node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:          <none>
```

csi-controller stateful set:
```
$ kubectl -n kube-system describe statefulset.apps/vsphere-csi-controller
Name:               vsphere-csi-controller
Namespace:          kube-system
CreationTimestamp:  Mon, 02 Mar 2020 16:09:06 -0800
Selector:           app=vsphere-csi-controller
Labels:             <none>
Annotations:        kubectl.kubernetes.io/last-applied-configuration:
                      {"apiVersion":"apps/v1","kind":"StatefulSet","metadata":{"annotations":{},"name":"vsphere-csi-controller","namespace":"kube-system"},"spec...
Replicas:           1 desired | 1 total
Update Strategy:    RollingUpdate
Pods Status:        1 Running / 0 Waiting / 0 Succeeded / 0 Failed
Pod Template:
  Labels:           app=vsphere-csi-controller
                    role=vsphere-csi
  Service Account:  vsphere-csi-controller
  Containers:
   csi-attacher:
    Image:      quay.io/k8scsi/csi-attacher:v1.1.1
    Port:       <none>
    Host Port:  <none>
    Args:
      --v=4
      --timeout=300s
      --csi-address=$(ADDRESS)
    Environment:
      ADDRESS:  /csi/csi.sock
    Mounts:
      /csi from socket-dir (rw)
   vsphere-csi-controller:
    Image:      gcr.io/cloud-provider-vsphere/csi/release/driver:v1.0.2
    Port:       9808/TCP
    Host Port:  0/TCP
    Args:
      --v=5
    Liveness:  http-get http://:healthz/healthz delay=10s timeout=3s period=5s #success=1 #failure=3
    Environment:
      CSI_ENDPOINT:        unix:///var/lib/csi/sockets/pluginproxy/csi.sock
      X_CSI_MODE:          controller
      VSPHERE_CSI_CONFIG:  /etc/cloud/csi-vsphere.conf
    Mounts:
      /etc/cloud from vsphere-config-volume (ro)
      /var/lib/csi/sockets/pluginproxy/ from socket-dir (rw)
   liveness-probe:
    Image:      quay.io/k8scsi/livenessprobe:v1.1.0
    Port:       <none>
    Host Port:  <none>
    Args:
      --csi-address=$(ADDRESS)
    Environment:
      ADDRESS:  /var/lib/csi/sockets/pluginproxy/csi.sock
    Mounts:
      /var/lib/csi/sockets/pluginproxy/ from socket-dir (rw)
   vsphere-syncer:
    Image:      gcr.io/cloud-provider-vsphere/csi/release/syncer:v1.0.2
    Port:       <none>
    Host Port:  <none>
    Args:
      --v=2
    Environment:
      FULL_SYNC_INTERVAL_MINUTES:  30
      VSPHERE_CSI_CONFIG:          /etc/cloud/csi-vsphere.conf
    Mounts:
      /etc/cloud from vsphere-config-volume (ro)
   csi-provisioner:
    Image:      quay.io/k8scsi/csi-provisioner:v1.2.2
    Port:       <none>
    Host Port:  <none>
    Args:
      --v=4
      --timeout=300s
      --csi-address=$(ADDRESS)
      --feature-gates=Topology=true
      --strict-topology
    Environment:
      ADDRESS:  /csi/csi.sock
    Mounts:
      /csi from socket-dir (rw)
  Volumes:
   vsphere-config-volume:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  vsphere-config-secret
    Optional:    false
   socket-dir:
    Type:          HostPath (bare host directory volume)
    Path:          /var/lib/csi/sockets/pluginproxy/csi.vsphere.vmware.com
    HostPathType:  DirectoryOrCreate
Volume Claims:     <none>
Events:            <none>
```

Controller Log:
```
I0303 21:31:17.384156       1 config.go:261] GetCnsconfig called with cfgPath: /etc/cloud/csi-vsphere.conf
I0303 21:31:17.384464       1 config.go:206] Initializing vc server ushil1vca001rd.gvservice.com
I0303 21:31:17.384516       1 controller.go:67] Initializing CNS controller
I0303 21:31:17.384543       1 virtualcentermanager.go:63] Initializing defaultVirtualCenterManager...
I0303 21:31:17.384574       1 virtualcentermanager.go:65] Successfully initialized defaultVirtualCenterManager
I0303 21:31:17.384601       1 virtualcentermanager.go:107] Successfully registered VC "ushil1vca001rd.gvservice.com"
I0303 21:31:17.384648       1 manager.go:62] Initializing volume.volumeManager...
I0303 21:31:17.384679       1 manager.go:66] volume.volumeManager initialized
time="2020-03-03T21:31:41Z" level=info msg="received signal; shutting down" signal=terminated
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x128 pc=0x868057]
goroutine 8 [running]:
google.golang.org/grpc.(*Server).GracefulStop(0x0)
	/go/pkg/mod/google.golang.org/grpc@v1.23.0/server.go:1393 +0x37
github.com/rexray/gocsi.(*StoragePlugin).GracefulStop.func1()
	/go/pkg/mod/github.com/rexray/gocsi@v1.0.0/gocsi.go:333 +0x35
sync.(*Once).Do(0xc00049808c, 0xc00020cef8)
	/usr/local/go/src/sync/once.go:44 +0xb3
github.com/rexray/gocsi.(*StoragePlugin).GracefulStop(0xc000498000, 0x211e8c0, 0xc000056018)
	/go/pkg/mod/github.com/rexray/gocsi@v1.0.0/gocsi.go:332 +0x56
github.com/rexray/gocsi.Run.func3()
	/go/pkg/mod/github.com/rexray/gocsi@v1.0.0/gocsi.go:121 +0x4e
github.com/rexray/gocsi.trapSignals.func1(0xc000194420, 0xc00045e900, 0xc00045e8d0)
	/go/pkg/mod/github.com/rexray/gocsi@v1.0.0/gocsi.go:502 +0x143
created by github.com/rexray/gocsi.trapSignals
	/go/pkg/mod/github.com/rexray/gocsi@v1.0.0/gocsi.go:487 +0x107
```

Syncer Log:
```
I0303 20:06:37.295500       1 virtualcentermanager.go:63] Initializing defaultVirtualCenterManager...
I0303 20:06:37.295545       1 virtualcentermanager.go:65] Successfully initialized defaultVirtualCenterManager
I0303 20:06:37.295561       1 virtualcentermanager.go:107] Successfully registered VC "ushil1vca001rd.gvservice.com"
I0303 20:07:07.423800       1 kubernetes.go:56] k8s client using in-cluster config
I0303 20:07:07.425181       1 metadatasyncer.go:60] FullSync: fullSync interval is set to 30 minutes
I0303 20:07:07.425394       1 metadatasyncer.go:159] Initialized metadata syncer
I0303 20:37:07.425977       1 metadatasyncer.go:124] fullSync is triggered
I0303 20:37:07.426298       1 fullsync.go:38] FullSync: start
I0303 20:37:07.430248       1 manager.go:62] Initializing volume.volumeManager...
I0303 20:37:07.430394       1 manager.go:66] volume.volumeManager initialized
I0303 20:37:07.526798       1 fullsync.go:97] FullSync: end
I0303 21:07:07.425419       1 metadatasyncer.go:124] fullSync is triggered
I0303 21:07:07.425739       1 fullsync.go:38] FullSync: start
I0303 21:07:07.532262       1 fullsync.go:97] FullSync: end
```

I have this exact configuration working on a separate cluster, though if I restart that cluster's csi-controller, it will go back to broken for a while.

--- a/go.mod
+++ b/go.mod
@@ -47,7 +47,7 @@ require (
 	github.com/prometheus/client_golang v1.1.0 // indirect
 	github.com/prometheus/client_model v0.0.0-20190812154241-14fe0d1b01d4 // indirect
 	github.com/prometheus/procfs v0.0.4 // indirect
-	github.com/rexray/gocsi v1.0.0
+	github.com/rexray/gocsi v1.0.1
 	github.com/sirupsen/logrus v1.4.2 // indirect
 	github.com/soheilhy/cmux v0.1.4 // indirect
 	github.com/spf13/cobra v0.0.5 // indirect

